PUBLISH.md

```diff
@@ -0,0 +1,88 @@
+# Publishing packages
+
+
+## CXL Registry
+
+We maintain custom npm registry: https://npm.cxl.co and publish there Aybolit packages.
+
+It is important to check if there's `.npmrc` file on main Aybolit directory with following entry:
+
+```
+@conversionxl:registry=https://npm.cxl.co
+```
+
+## Log in
+
+In order to publish packages you need to log in to npm registry.
+
+```
+npm login --registry=https://npm.cxl.co
+```
+
+## Install dependencies and build
+
+Run following commands in main repository directory:
+
+```
+yarn
+yarn build
+yarn dist
+```
+
+## Publish package
+
+**Important:** before performing following steps make sure your workdir is clean and contains only necessary files and directories. Clean it up from everyting else as publish everything sitting in the package directory.
+
+In following steps `cxl-ui` package is used but you can replace it with name of the other package.
+
+
+### Steps
+
+1. `cd packages/cxl-ui`
+2. Edit package version in `package.json`.
+
+  ```diff
+  - "version": "1.1.0-alpha.2",
+  + "version": "1.1.0-alpha.3",
+  ```
+
+3. Save and commit it.
+4. `yarn publish`
+5. Go through all steps. When asked about version, use value you just placed in `package.json` eg. `1.1.0-alpha.3`
+6. Done.
+
+### yarn options
+
+It's possible to use following options:
+
+* `--registry <url> // override configuration registry`
+* `--new-version [version] // new version`
+* `--tag [tag] // tag`
+
+
+More: `yarn --help`
+
+### Tags
+
+Package versions can be tagged, eg. `cxl-ui` has `playbooks` tag.
+
+#### Flow for new tag
+
+When you publish, simply add `--tag tagname` param to the publish command.
+
+#### Flow if the tag exists
+
+If you want tag to point always to latest version of specific branch (?) (eg. `playbooks`) you need remove tag first and then tag latest release.
+
+1. `npm dist-tag rm <pkg> <tag> --registry=https://npm.cxl.co`
+2. `npm dist-tag add <pkg>@<version> [<tag>]`
+
+More:
+
+* [npm dist tag](https://docs.npmjs.com/cli/v6/commands/npm-dist-tag)
+* [yarn tag](https://classic.yarnpkg.com/en/docs/cli/tag/)
+
+## Publish flow and standards
+
+* every package has to be published separately (`cxl-ui`, `cxl-lumo-styles`, ...),
+* we keep versions of the packages in sync, even if changes are introduced to only one of them.
```

Comment:
Instead of modifying `packages/cxl-ui/package.json` directly, we could use `lerna version` which would update all of the package versions keeping them in sync.

https://github.com/lerna/lerna/tree/main/commands/version#readme

Reply:
Doesn't 
`yarn publish` take care of versions by triggering `lerna version` behind the schenes?

Reply:
`lerna publish` does run `lerna version` behind the scenes:

```
lerna notice cli v3.22.1 
lerna info current version 0.1.0
lerna info Looking for changed packages since v0.1.0
? Select a new version (currently 0.1.0) Prepatch (0.1.1-alpha.0)

Changes:
 - @aybolit/core: 0.1.0 => 0.1.1-alpha.0
 - @conversionxl/cxl-lumo-styles: 1.0.0 => 0.1.1-alpha.0
 - @conversionxl/cxl-ui: 1.0.0 => 0.1.1-alpha.0
 - @conversionxl/sass-render: 0.2.0 => 0.1.1-alpha.0 (private)
 - @conversionxl/storybook: 1.0.0 => 0.1.1-alpha.0 (private)
 - @aybolit/test-helpers: 0.1.0 => 0.1.1-alpha.0 (private)

? Are you sure you want to publish these packages? No
```

I tried running `yarn publish` and got:

```
Package marked as private, not publishing.
```

Reply:
`lerna` in fixed mode will grab the version from `lerna.json`, and bump all packages from there.